action.yml

```
@@ -205,6 +250,39 @@ runs:
       path: .tmp-ansible-collection-checkout
       persist-credentials: false
       ref: ${{ inputs.git-checkout-ref }}
+      fetch-depth: >-
+        ${{
+          steps.compute-change-detection-coverage.outputs.change-detection-arg
+          && '0' || '1'
+        }}
+
+  - name: Log the next step action
+    if: >-
+      !inputs.collection-src-directory
+      && steps.compute-change-detection-coverage.outputs.change-detection-arg
+    run: >-
+      echo ▷ Create branches for change detection
+    shell: bash
+  - name: Create branches for change detection
+    if: >-
+      !inputs.collection-src-directory
+      && steps.compute-change-detection-coverage.outputs.change-detection-arg
+    run: |
+      # Create a branch for the current HEAD, which happens to be a merge commit
+      git checkout -b __action_branch_name
+
+      # Check out the target branch and name it
+      git checkout -b ${{
+        github.event.pull_request.base.ref
+      }} --track origin/${{
+        github.event.pull_request.base.ref
+      }}
+
+      # Check out the merge commit
+      git checkout __action_branch_name
```

Why is this branch name needed? Just so you could switch branches temporarily?
Can it just be something like
```suggestion
      # Check out the target branch and name it
      git branch ${{
        github.event.pull_request.base.ref
      }} --track origin/${{
        github.event.pull_request.base.ref
      }}
```

---

ansible-test will die if `HEAD` is not part of a branch (maybe a tag will also work, but a branch is more common).

---

I'm pretty sure my variant does the same, though. Have you tested it?

---

Your suggestion does not give a name to current `HEAD`, only to the target branch. It will not work without the `git checkout -b __action_branch_name` line (or something equivalent, the name of the branch doesn't matter as long as there is a branch).

---

I've applied an adjusted version of your suggestion in b40aa9fb5789751da26b476b7d5601147d704292.

---

So since you're creating the branch on that commit already, it doesn't seem like `git checkout` is necessary.